src/instance.ts

```diff
@@ -47,77 +54,57 @@ export default class Instance {
         this.platformVersion = splitInstance[0];
         this.cluster = splitInstance[1];
         this.id = splitInstance[2];
-
+        
         this.serviceName = options.serviceName;
         this.serviceVersion = options.serviceVersion;
-        this.tokenProvider = options.tokenProvider;
-
-        if(options.host){
-            this.host = options.host;
-        } else{
-            this.host = `${this.cluster}.${HOST_BASE}`;
-        }
-
+        
+        this.host = options.host || `${this.cluster}.${HOST_BASE}`;
+        this.logger = options.logger || new ConsoleLogger();
+        
         this.client = options.client || new BaseClient({
             encrypted: options.encrypted,
-            host: this.host
+            host: this.host,
+            logger: this.logger
         });
-        
-        if(options.logger !== undefined){
-            this.logger = options.logger;
-        }
-        else{
-            this.logger = new ConsoleLogger();
-        }
     }
+    
 
-    request(options: RequestOptions): Promise<any> {
+    request<T>(options: RequestOptions): NetworkResponse<T>{
         options.path = this.absPath(options.path);
-        const tokenProvider = options.tokenProvider || this.tokenProvider;
-        if (!options.jwt && tokenProvider) {
-            return tokenProvider.fetchToken().then((jwt) => {
-                return this.client.request({ jwt, ...options });
-            });
-        } else {
-            return this.client.request(options);
-        }
+        return this.client.request<T>(options);
     }
 
-    subscribe(options: SubscribeOptions): Subscription {
-        options.path = this.absPath(options.path);
-        options.logger = this.logger;
-
-        let subscription: Subscription = this.client.newSubscription(options);
-
-        const tokenProvider = options.tokenProvider || this.tokenProvider;
-        if (options.jwt) {
-            subscription.open(options.jwt);
-        } else if (tokenProvider) {
-            tokenProvider.fetchToken().then((jwt) => {
-                subscription.open(jwt);
-            }).catch((err) => {
-                subscription.unsubscribe(err);
-            });
-        } else {
-            subscription.open(null);
-        }
-        return subscription;
+    subscribe(options: SubscribeOptions): Subscription { 
```

Comment:
Would it be worth making it symmetric with the base client and calling it `subscribeNonResuming`?